rbac/internal/views.py

```diff
@@ -1150,3 +1151,16 @@ def principal_removal(request):
                 bootstrap_service.update_user(user)
 
         return HttpResponse(f"Users deleted: {principal_usernames}", status=204)
+
+
+def retrieve_ungrouped_workspace(request, org_id):
+    """GET or create ungrouped workspace for HBI."""
+    if request.method != "GET":
+        return HttpResponse("Invalid request method, only GET is allowed.", status=405)
+
+    ungrouped_hosts = Workspace.objects.filter(tenant__org_id=org_id, type=Workspace.Types.UNGROUPED_HOSTS).first()
+    if not ungrouped_hosts:
+        bootstrap_service = V2TenantBootstrapService(OutboxReplicator())
```

Comment:
@astrozzc do we count that org is always bootstrapped ? or should bootstrap it for certitude ?

Reply:
Right, the tenant should always be bootstrapped currently